nucypher-core/src/message_kit.rs

```diff
@@ -40,7 +41,7 @@ impl MessageKit {
         Self {
             capsule,
             ciphertext,
-            conditions: conditions.map(|c| c.into()),
+            conditions: conditions.cloned(),
```

Comment:
Hmm... not obvious to me what's happening here.

Reply:
Formerly `conditions` were of type `Option<&[u8]>`, and we needed to convert them to `Option<Box<[u8]>>` - hence the `map` (to get inside the `Option`) and the conversion from `&[u8]` (using `impl From<&[u8]> for Box<[u8]>`).

After the changes, the input is `Option<&Conditions>`, and we need to convert it to `Option<Conditions>`. We could use `conditions.map(|c| c.clone())`, but [`cloned()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.cloned) is a shortcut for it.